File: solutions/clojure/leap/3/src/leap.clj
(ns leap)

(defn is-divisible-by? [num div]
  (zero? (mod num div))
)

(defn leap-year? [year]
  (and 
     (is-divisible-by? year 4)
     (or  
         (not (is-divisible-by? year 100))
         (is-divisible-by? year 400)
     )
  )
)

File: solutions/csharp/leap/3/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year) =>
        year.IsDivisibleBy(400)
        || year.IsDivisibleBy(4)
            && !year.IsDivisibleBy(100);

    static bool IsDivisibleBy(this int number, int divisor) => number % divisor == 0;
}
